### PR TITLE
Issue #KN-802 feat: Sync QR Image to DIAL code ES index

### PR DESCRIPTION
**Pull request by Jayaprakash8887**

### CODE REVIEW

File: app/controllers/dialcode/DialcodeV3Controller.java [hunk 1]
```diff
@@ -76,7 +76,12 @@ public Promise<Result> searchDialCode() {
         Request request = getRequest();
         try {
             Map<String, Object> map = (Map<String, Object>) request.get(DialCodeEnum.search.name());
-            Response response = dialCodeManager.searchDialCode(request.getContext(), map);
+            List<String> fieldsList = null;
+            if(request.get(DialCodeEnum.fields.name()) != null) {
+                fieldsList = (List<String>) request.get(DialCodeEnum.fields.name());
+                fieldsList.add("identifier");
+            }
```
Comment (maheshkumargangula) [inline app/controllers/dialcode/DialcodeV3Controller.java:83]: What if you get a string in "fields"?

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: app/controllers/dialcode/DialcodeV3Controller.java (hunk 1 above)
Comment (Jayaprakash8887) [reply app/controllers/dialcode/DialcodeV3Controller.java:83]: That is returning ClassCastException. Added this code as a good to have. If this is not required, will remove the same.

Comment (maheshkumargangula) [reply app/controllers/dialcode/DialcodeV3Controller.java:83]: What is the API returning when it throws ClassCastException?

Comment (Jayaprakash8887) [reply app/controllers/dialcode/DialcodeV3Controller.java:83]: image

Comment (maheshkumargangula) [reply app/controllers/dialcode/DialcodeV3Controller.java:83]: We should handle this scenario and return client error or accept the string format and covert to list internally. It should not return an Internal Server Error. Please handle.

File: app/controllers/dialcode/DialcodeV3Controller.java [hunk 2]
```diff
@@ -76,7 +76,16 @@ public Promise<Result> searchDialCode() {
...
             Map<String, Object> map = (Map<String, Object>) request.get(DialCodeEnum.search.name());
-            Response response = dialCodeManager.searchDialCode(request.getContext(), map);
+            List<String> fieldsList = new ArrayList<String>();
+            if(request.get(DialCodeEnum.fields.name()) != null) {
+                try {
+                    fieldsList = (List<String>) request.get(DialCodeEnum.fields.name());
+                } catch (ClassCastException ce) {
+                    fieldsList.add(request.get(DialCodeEnum.fields.name()).toString());
+                }
+                fieldsList.add("identifier");
```
Comment (maheshkumargangula) [inline app/controllers/dialcode/DialcodeV3Controller.java:86]: This code doesn't work if we give `fields` as Map. It should not give 500 error. Please handle this also...

**Review by maheshkumargangula — APPROVED** [review]

**Pull request merged**